src/library/scala/collection/View.scala

```diff
@@ -163,50 +163,18 @@ object View extends IterableFactory[View] {
     override def isEmpty: Boolean = underlying.isEmpty
   }
 
-  /** A class that partitions an underlying collection into two views */
-  @SerialVersionUID(3L)
-  class Partition[A](val underlying: SomeIterableOps[A], val p: A => Boolean) extends Serializable {
-
-    /** The view consisting of all elements of the underlying collection
-     *  that satisfy `p`.
-     */
-    val first = new Partitioned(this, true)
-
-    /** The view consisting of all elements of the underlying collection
-     *  that do not satisfy `p`.
-     */
-    val second = new Partitioned(this, false)
-  }
-
   /** A view representing one half of a partition. */
   @SerialVersionUID(3L)
-  class Partitioned[A](partition: Partition[A], cond: Boolean) extends AbstractView[A] {
-    def iterator = partition.underlying.iterator.filter(x => partition.p(x) == cond)
-    override def knownSize: Int = if (partition.underlying.knownSize == 0) 0 else super.knownSize
+  class Partitioned[A](val underlying: SomeIterableOps[A], val p: A => Boolean, cond: Boolean) extends AbstractView[A] {
+    def iterator = underlying.iterator.filter(x => p(x) == cond)
+    override def knownSize: Int = if (underlying.knownSize == 0) 0 else super.knownSize
     override def isEmpty: Boolean = iterator.isEmpty
   }
 
-  /** A class that splits an underlying collection into two views */
   @SerialVersionUID(3L)
-  class PartitionWith[A, A1, A2](val underlying: SomeIterableOps[A], val f: A => Either[A1, A2]) extends Serializable {
-
-    /** The view consisting of all elements of the underlying collection
-      *  that map to `Left`.
-      */
-    val left: View[A1] = new LeftPartitionedWith(this, f)
-
-
-    /** The view consisting of all elements of the underlying collection
-      *  that map to `Right`.
-      */
-    val right: View[A2] = new RightPartitionedWith(this, f)
-
-  }
-
-  @SerialVersionUID(3L)
-  class LeftPartitionedWith[A, A1, A2](partitionWith: PartitionWith[A, A1, A2], f: A => Either[A1, A2]) extends AbstractView[A1] {
+  class LeftPartitionedWith[A, A1, A2](val underlying: SomeIterableOps[A], f: A => Either[A1, A2]) extends AbstractView[A1] {
```

Comment:
Any reason for `underlying` being public?

Reply:
@allanrenucci `underlying` is not being publicly accessed via this class, so I've removed the explicit `val` modifier now. Added `val` earlier keyword cause the `underlying` was marked with `val` from calling method, and I didn't know it also causes the field to be publicly accessible.